### PR TITLE
Add configurable timeout flag for kubernetes kubetctl drain command

### PR DESCRIPTION
**Pull request by neil-hickey**
**What this PR does / why we need it**:
+ This allows an operator to specify a timeout to force drain a node (at the kubectl level, not bosh)
+ It defaults to 0 to retain current functionality (never timeout)

**How can this PR be verified?**
When running an upgrade which has this helm chart deployed:

https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress

The drain will timeout after x time (if configured). Left at zero, this verifies that the upgrade will hang forever.

**Is there any change in kubo-deployment?**
None required. A change can be made if the CFCR team finds value in setting a non-zero default.

**Is there any change in kubo-ci?**
None required

**Does this affect upgrade, or is there any migration required?**
This will allow an operator to chose whether or not to timeout a kubectl node drain, so an upgrade will not get stuck infinitely.

**Which issue(s) this PR fixes:**
When upgrading CFCR, if a workload has a PodDisruptionBudget of 1 for a node, an upgrade will hang forever as kubernetes will not drain the node - Respecting the PodDisruptionBudget. 

An example of such a workload is the nginx ingress controller helm chart:

https://github.com/kubernetes/charts/tree/master/stable/nginx-ingress

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by alex-slynko** [comment]
@neil-hickey are there any sensible defaults? 
I don't feel comfortable to have this issue be in CFCR by default.

**Comment by alex-slynko** [comment]
@neil-hickey I think the default should not be infinite if it causes problems.

**Comment by st3v** [comment]
@alex-slynko @neil-hickey - I believe the default should be infinite as this is the default behaviour when running `drain` without setting a `timeout` flag at all. That default has been chosen for the `drain` command in order to guarantee uptime for all types of workloads, including the ones with PodDisruptionBudgets. IMO an operator should opt into having a timeout, we should not make that decision for them as it requires knowing the workloads running on the cluster.

**Comment by seanos11** [comment]
@neil-hickey we have looked at this, we have a few questions, especially on the desired behaviour for an operator

**Comment by neil-hickey** [comment]
sure @seanos11 . Feel free to slack me.

**Comment by JasonMorgan** [comment]
Hey folks, is there any movement on this? I'm regularly running into failures with the drain script running on my instances.

**Comment by neil-hickey** [comment]
Hey @CFCR is this something you are all considering still? I think this PR could be updated to include tests for the specs (I wasn't aware at the time of the testing strategy)

File: jobs/kubelet/templates/bin/drain.erb [hunk 1]
```diff
@@ -24,7 +24,7 @@ main() {
 
 drain_kubelet() {
   local node_name=$(kubectl --kubeconfig /var/vcap/jobs/kubelet/config/kubeconfig-drain get nodes -o wide -L spec.ip | grep "<%= spec.ip %>" | awk '{print $1}')
-  kubectl --kubeconfig /var/vcap/jobs/kubelet/config/kubeconfig-drain drain "${node_name}" --grace-period 10 --force --delete-local-data  --ignore-daemonsets
+  kubectl --kubeconfig /var/vcap/jobs/kubelet/config/kubeconfig-drain drain "${node_name}" --grace-period 10 --force --delete-local-data  --ignore-daemonsets --timeout <% p("kubectl-drain-timeout" %>
```
Comment (iainsproat) [inline jobs/kubelet/templates/bin/drain.erb]: There's a closing parenthesis missing in the property.

**Comment by iainsproat** [comment]
@neil-hickey myself and @karampok have asked for this to be discussed in pre-IPM.

I've left a comment for your action.

**Pull request merged**